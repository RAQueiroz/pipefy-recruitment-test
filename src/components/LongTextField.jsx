import React from "react";
import styled from "styled-components";
import StyledLabel from "./StyledLabel";
import { InlineError } from "./StyledElements";
import Colors from "../utils/Colors";

const StyledInput = styled.textarea`
  border: solid 1px ${props => (props.error ? Colors.errorLight : "#ccc")};
  line-height: 40px;
  padding-left: 10px;
  border-radius: 5px;
`;

const LongTextField = ({
  label,
  id,
  error,
  value = "",
  onChange = () => {}
}) => (
  <div>
    <StyledLabel>
      {label}
      <StyledInput
        value={value}
        type="text"
        id={id}
        onChange={({ target }) => onChange({ id, value: target.value })}
      />

    </StyledLabel>
      {error && <InlineError>{error}</InlineError>}
  </div>
);

export default LongTextField;
