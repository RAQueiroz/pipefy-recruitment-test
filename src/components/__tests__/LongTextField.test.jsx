import React from "react";
import LongTextField from "../LongTextField";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe(" <LongTextField/>", () => {
  const onChangeSpy = jest.fn();

  const item = {
    id: "xpto",
    label: "Name",
    value: "Josh"
  };

  const wrapper = shallow(<LongTextField onChange={onChangeSpy} {...item} />);

  it("renders", () => {
    expect(wrapper.find(`#${item.id}`)).toHaveLength(1);
  });

  it("to match snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("display label based on props", () => {
    expect(wrapper.containsMatchingElement(item.label)).toBeTruthy();
    wrapper.setProps({ label: "Surname" });
    expect(wrapper.containsMatchingElement("Surname")).toBeTruthy();
  });

  it("display value based on props", () => {
    expect(wrapper.find(`#${item.id}`).props().value).toEqual(item.value);
    wrapper.setProps({ value: "John" });
    expect(wrapper.find(`#${item.id}`).props().value).toEqual("John");
  });

  it("check if onChange works", () => {
    wrapper
      .find(`#${item.id}`)
      .simulate("change", { target: { value: "Hello" } });
    expect(onChangeSpy).toBeCalledWith({ id: "xpto", value: "Hello" });
  });
});
