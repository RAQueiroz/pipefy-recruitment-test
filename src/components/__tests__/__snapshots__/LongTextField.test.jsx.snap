// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` <LongTextField/> to match snapshot 1`] = `
<div>
  <styled.label>
    Name
    <styled.textarea
      id="xpto"
      onChange={[Function]}
      type="text"
      value="Josh"
    />
  </styled.label>
</div>
`;
